examples/federated-learning/run.py

```diff
@@ -6,156 +8,187 @@
+      y_hat, loss = self._build_validation_step(x, y)
+
+      with tf.control_dependencies([update_op]):
+        return tf.print('expect', loss, y, y_hat, summarize=50)
 
 
 class DataOwner:
-
-    BATCH_SIZE = 30
-
-    def __init__(self, player_name, build_training_model):
-        self.player_name = player_name
-        self._build_training_model = build_training_model
-
-    def _build_data_pipeline(self):
-
-        def normalize(image, label):
-            image = tf.cast(image, tf.float32) / 255.0
-            return image, label
-
-        dataset = tf.data.TFRecordDataset(["./data/train.tfrecord"])
-        dataset = dataset.map(decode)
-        dataset = dataset.map(normalize)
-        dataset = dataset.repeat()
-        dataset = dataset.batch(self.BATCH_SIZE)
-
-        iterator = dataset.make_one_shot_iterator()
-        return iterator.get_next()
-
-    def compute_gradient(self):
-
-        with tf.name_scope('data_loading'):
-            x, y = self._build_data_pipeline()
-
-        with tf.name_scope('gradient_computation'):
-            grads = self._build_training_model(x, y)
-
-        return grads
-
-
-model_owner = ModelOwner('model-owner')
-data_owners = [
-    DataOwner('data-owner-0', model_owner.build_training_model),
-    DataOwner('data-owner-1', model_owner.build_training_model),
-    DataOwner('data-owner-2', model_owner.build_training_model),
-]
-
-model_grads = zip(*(
-    tfe.define_private_input(data_owner.player_name, data_owner.compute_gradient)
-    for data_owner in data_owners
-))
-
-with tf.name_scope('secure_aggregation'):
+  """Contains methods meant to be executed by a data owner.
+
+  Args:
+    player_name: `str`, name of the `tfe.player.Player`
+                 representing the data owner
+    build_update_step: `Callable`, the function used to construct
+                       a local federated learning update.
+  """
+
+  BATCH_SIZE = 30
+
+  def __init__(self, player_name, build_update_step):
+    self.player_name = player_name
+    self._build_update_step = build_update_step
+
+  def _build_data_pipeline(self):
+    """Build local data pipeline for federated DataOwners."""
+    def normalize(image, label):
+      image = tf.cast(image, tf.float32) / 255.0
+      return image, label
+
+    dataset = tf.data.TFRecordDataset(["./data/train.tfrecord"])
+    dataset = dataset.map(decode)
+    dataset = dataset.map(normalize)
+    dataset = dataset.repeat()
+    dataset = dataset.batch(self.BATCH_SIZE)
+
+    iterator = dataset.make_one_shot_iterator()
+    return iterator.get_next()
+
+  def compute_gradient(self):
+    """Compute gradient given current model parameters and local data."""
+    with tf.name_scope('data_loading'):
+      x, y = self._build_data_pipeline()
+
+    with tf.name_scope('gradient_computation'):
+      grads = self._build_update_step(x, y)
+
+    return grads
+
+
+if __name__ == "__main__":
+  model_owner = ModelOwner('model-owner')
+  data_owners = [
```

Comment:
tiny thing but there seems to be too much indentation here

Reply:
see response below